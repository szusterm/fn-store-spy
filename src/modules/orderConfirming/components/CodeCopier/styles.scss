@import '~material-colors/dist/colors';
@import '~scss/_opacity-colors';
@import '~scss/_transition';

.code-copier {
	width: 100%;
	padding: 15px;

	&--code {
		cursor: pointer;

		padding: 8px;
		font-size: 3.5em;
		font-weight: 700;
		letter-spacing: 0.1em;
		text-align: center;
		text-transform: uppercase;

		background-color: $md-blue-500;
		border-radius: 15px;

		@include white-color(0.95);
		@include transition(sh);

		&:hover, &:focus {
			background-color: $md-blue-600;

			@include white-color(1);
		}

		&::selection {
			@include white-background(0.9);
			color: $md-blue-400;
		}

		&--copied {
			background-color: $md-green-500;

			&::selection {
				color: $md-green-400;
			}

			&:hover, &:focus {
				background-color: $md-green-600;
			}
		}
	}

	&--info {
		padding: 8px;
		text-align: center;
		text-transform: uppercase;
		font-weight: 700;

		.info {
			&--how-to {
				@include black-color(0.3);
			}

			&--copied {
				color: $md-green-300;
			}
		}
	}
}