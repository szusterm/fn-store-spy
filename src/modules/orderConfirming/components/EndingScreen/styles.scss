@import '~scss/_opacity-colors';
@import '~scss/_transition';
@import '~scss/_center-items';

.ending-screen {
	@include center-items();

	position: fixed;
	top: 0;
	bottom: 0;
	left: 0;
	right: 0;
	z-index: 100;

	@include white-background(1);
	@include transition(sh);
	transition-property: opacity;

	&--hidden {
		transform: translate(100%, 0);
		opacity: 0;
	}
}