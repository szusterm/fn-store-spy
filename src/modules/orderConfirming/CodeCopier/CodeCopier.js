import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './styles.scss';

export class CodeCopier extends Component {
	render() {
		return (
			<input
				className={'code-copier'}
			/>
		);
	}
}

CodeCopier.propTypes = {
	code: PropTypes.string //redux
};

const mapStateToProps = (state) => {
	const {code} = state.orderConfirming;

	return {code};
};

export default connect(mapStateToProps, null)(CodeCopier);