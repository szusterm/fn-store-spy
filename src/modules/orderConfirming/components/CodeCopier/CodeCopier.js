import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {CopyToClipBoard} from 'react-copy-to-clipboard';

import './styles.scss';

export class CodeCopier extends Component {
	render() {
		const {code} = this.props;

		return (
			<input
				className={'code-copier'}
				readOnly={true}
				value={code}
			/>
		);
	}
}

CodeCopier.propTypes = {
	code: PropTypes.string //redux
};

const mapStateToProps = (state) => {
	const {code} = state.orderConfirming;

	return {code};
};

export default connect(mapStateToProps, null)(CodeCopier);