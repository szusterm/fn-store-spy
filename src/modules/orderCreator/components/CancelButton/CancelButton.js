import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {closeOrderList} from '../../actions';

import './styles.scss';

export class CancelButton extends Component {
	render() {
		const {closeOrderList} = this.props;

		return (
			<div className={'col-8'}>
				<button
					className={'cancel-button'}
					onClick={closeOrderList}
				>
					<span className={'material-icons'}>close</span>
				</button>
			</div>
		);
	}
}

CancelButton.propTypes = {
	closeOrderList: PropTypes.func //redux
};

const mapDispatchToProps = {closeOrderList};

export default connect(null, mapDispatchToProps)(CancelButton);