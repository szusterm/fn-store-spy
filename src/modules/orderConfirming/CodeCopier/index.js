import CodeCopier from './CodeCopier';

export default CodeCopier;