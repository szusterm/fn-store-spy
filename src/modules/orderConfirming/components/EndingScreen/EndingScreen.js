import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {hideEndingScreen} from '../../actions';

import CodeCopier from '../CodeCopier';
import BlackCurtain from '../../../../components/BlackCurtain';
import MessengerLink from '../../../../components/MessengerLink';
import CancelButton from '../../../../components/CancelButton';

import './styles.scss';

export class EndingScreen extends Component {
	render() {
		const {endingScreenOpened, hideEndingScreen} = this.props;

		return (
			<div>
				<BlackCurtain
					show={endingScreenOpened}
					onClick={hideEndingScreen}
				/>
				<div className={`ending-screen ${(!endingScreenOpened) ? 'ending-screen--hidden' : ''}`}>
					<div className={'ending-screen--prompt'}>
						<h2 className={'prompt--text'}>Send this code to the bot</h2>
					</div>
					<div className={'ending-screen--code-copier-box'}>
						<CodeCopier/>
					</div>
					<div className={'ending-screen--buttons'}>
						<MessengerLink/>
						<CancelButton onClick={hideEndingScreen}/>
					</div>
				</div>
			</div>
		);
	}
}

EndingScreen.propTypes = {
	endingScreenOpened: PropTypes.bool, //redux
	hideEndingScreen: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {endingScreenOpened} = state.orderConfirming;

	return {endingScreenOpened};
};

const mapDispatchToProps = {hideEndingScreen};

export default connect(mapStateToProps, mapDispatchToProps)(EndingScreen);