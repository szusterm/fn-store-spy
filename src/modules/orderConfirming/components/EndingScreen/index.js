import EndingScreen from './EndingScreen';

export default EndingScreen;