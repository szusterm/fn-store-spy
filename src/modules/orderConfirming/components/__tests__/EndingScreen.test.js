import React from 'react';
import {shallow} from 'enzyme';

import CodeCopier from '../CodeCopier';
import CancelButton from '../../../../components/CancelButton';
import BlackCurtain from '../../../../components/BlackCurtain';

import {EndingScreen} from '../EndingScreen/EndingScreen';

const setup = (propOverrides) => {
	const props = Object.assign({
		endingScreenOpened: true, //redux
		hideEndingScreen: () => true
	}, propOverrides);

	const wrapper = shallow(<EndingScreen {...props}/>);

	return {props, wrapper};
};

describe('EndingScreen Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('hides, if endingScreenOpened is false', () => {
		const {wrapper} = setup();

		const hiddenEndingScreenClass = '.ending-screen--hidden';

		wrapper.setProps({endingScreenOpened: false});
		expect(wrapper.find(hiddenEndingScreenClass).exists()).toBe(true);

		wrapper.setProps({endingScreenOpened: true});
		expect(wrapper.find(hiddenEndingScreenClass).exists()).toBe(false);
	});

	it('hides if BlackCurtain calls onClick callback', () => {
		const {wrapper, props} = setup({
			hideEndingScreen: jest.fn()
		});

		wrapper.find(BlackCurtain).props().onClick();

		expect(props.hideEndingScreen).toHaveBeenCalledTimes(1);
	});

	it('puts listOpened from a redux state as a prop to BlackCurtain', () => {
		const {wrapper, props} = setup({
			endingScreenOpened: false
		});

		const showProp = wrapper.find(BlackCurtain).props().show;

		expect(showProp).toBe(props.endingScreenOpened);
	});

	it('hides if CancelButton calls onClick callback', () => {
		const {wrapper, props} = setup({
			hideEndingScreen: jest.fn()
		});

		wrapper.find(CancelButton).props().onClick();

		expect(props.hideEndingScreen).toHaveBeenCalledTimes(1);
	});

	it('renders CodeCopier', () => {
		const {wrapper} = setup();

		expect(wrapper.find(CodeCopier).exists()).toBe(true);
	});
});