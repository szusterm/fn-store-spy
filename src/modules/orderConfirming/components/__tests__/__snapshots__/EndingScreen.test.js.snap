// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndingScreen Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EndingScreen
    endingScreenOpened={true}
    hideEndingScreen={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BlackCurtain
          onClick={[Function]}
          show={true}
        />,
        <div
          className="ending-screen "
        >
          <div
            className="ending-screen--prompt"
          >
            <h2
              className="prompt--text"
            >
              Send this code to the bot
            </h2>
          </div>
          <div
            className="ending-screen--code-copier-box"
          >
            <Connect(CodeCopier) />
          </div>
          <div
            className="ending-screen--buttons"
          >
            <Connect(MessengerLink) />
            <CancelButton
              onClick={[Function]}
            />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
          "show": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ending-screen--prompt"
            >
              <h2
                className="prompt--text"
              >
                Send this code to the bot
              </h2>
            </div>,
            <div
              className="ending-screen--code-copier-box"
            >
              <Connect(CodeCopier) />
            </div>,
            <div
              className="ending-screen--buttons"
            >
              <Connect(MessengerLink) />
              <CancelButton
                onClick={[Function]}
              />
            </div>,
          ],
          "className": "ending-screen ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="prompt--text"
              >
                Send this code to the bot
              </h2>,
              "className": "ending-screen--prompt",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send this code to the bot",
                "className": "prompt--text",
              },
              "ref": null,
              "rendered": "Send this code to the bot",
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(CodeCopier) />,
              "className": "ending-screen--code-copier-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(MessengerLink) />,
                <CancelButton
                  onClick={[Function]}
                />,
              ],
              "className": "ending-screen--buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BlackCurtain
            onClick={[Function]}
            show={true}
          />,
          <div
            className="ending-screen "
          >
            <div
              className="ending-screen--prompt"
            >
              <h2
                className="prompt--text"
              >
                Send this code to the bot
              </h2>
            </div>
            <div
              className="ending-screen--code-copier-box"
            >
              <Connect(CodeCopier) />
            </div>
            <div
              className="ending-screen--buttons"
            >
              <Connect(MessengerLink) />
              <CancelButton
                onClick={[Function]}
              />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
            "show": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ending-screen--prompt"
              >
                <h2
                  className="prompt--text"
                >
                  Send this code to the bot
                </h2>
              </div>,
              <div
                className="ending-screen--code-copier-box"
              >
                <Connect(CodeCopier) />
              </div>,
              <div
                className="ending-screen--buttons"
              >
                <Connect(MessengerLink) />
                <CancelButton
                  onClick={[Function]}
                />
              </div>,
            ],
            "className": "ending-screen ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2
                  className="prompt--text"
                >
                  Send this code to the bot
                </h2>,
                "className": "ending-screen--prompt",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Send this code to the bot",
                  "className": "prompt--text",
                },
                "ref": null,
                "rendered": "Send this code to the bot",
                "type": "h2",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(CodeCopier) />,
                "className": "ending-screen--code-copier-box",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(MessengerLink) />,
                  <CancelButton
                    onClick={[Function]}
                  />,
                ],
                "className": "ending-screen--buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
